ui/pages/confirm-token-transaction-base/confirm-token-transaction-base.container.js

```diff
@@ -49,27 +49,51 @@ const mapStateToProps = (state, ownProps) => {
     hexMaximumTransactionFee,
   } = transactionFeeSelector(state, transaction);
   const tokens = getTokens(state);
+  const collectibles = getCollectibles(state);
+
   const currentToken = tokens?.find(({ address }) =>
     isEqualCaseInsensitive(tokenAddress, address),
   );
-  const { decimals, symbol: tokenSymbol } = currentToken || {};
+  const currentCollectible = collectibles?.find(({ address }) =>
```

Comment:
TODO, this needs to search by tokenId as well!

Reply:
done here https://github.com/MetaMask/metamask-extension/pull/13048/commits/13dd666d6959607aa0709d66b2257f7b067ea536